docs/en/advanced_config/tuning_the_ecl_ekf.md

```diff
@@ -348,6 +348,64 @@ The `hpos_drift_rate`, `vpos_drift_rate` and `hspd` are calculated over a period
 Note that `ekf2_gps_drift` is not logged!
 :::
 
+#### GNSS Fault Detection
+
+PX4's GNSS fault detection protects against malicious or erroneous GNSS signals using selective fusion control based on measurement validation.
+The detection logic depends on the [EKF2_GPS_MODE](../advanced_config/parameter_reference.md#EKF2_GPS_MODE) parameter, and operates differently for horizontal position and altitude measurements.
+
+##### Configuration
+
+The fault detection logic depends on the GPS mode, which is set using the [EKF2_GPS_MODE](../advanced_config/parameter_reference.md#EKF2_GPS_MODE) parameter:
+
+- **Automatic (`0`)** (Default): EKF2 resets on fusion timeouts if no other source of position is available.
+- **Dead-reckoning (`1`)**: EKF2 may reset if no other sources of position or velocity are available.
```

Comment:
This is my understanding 
- the normal GNSS trusting"" mode resets when the data can't be fused - but only if no other source of position is available. 
  - If you had another source you'd presumably stop using the GNSS and record a fault for it?
- In dead reckoning mode the EKF resets only if there are no other sources of position or velocity. The reasoning being that in this case you have a velocity so you can you can estimate position and discard GNSS data.

Is that right?

So to be clear 
- "resetting GNSS" is a bad thing? What is likely to happen in automatic mode?
- Faults are not a bad thing - they are for information of other systems - EKF just keeps on going if it has another source -r ight?